core/consensus/babe/impl/babe.cpp

```diff
@@ -389,62 +326,59 @@ namespace kagome::consensus::babe {
 
       {  // Fill disputes
         std::latch latch(1);
-        dispute_coordinator_->getDisputeForInherentData(
-            ctx.parent, [&](auto res) {
-              parachain_inherent_data.disputes = std::move(res);
-              latch.count_down();
-            });
+        dispute_coordinator_->getDisputeForInherentData(parent_, [&](auto res) {
+          parachain_inherent_data.disputes = std::move(res);
+          latch.count_down();
+        });
         latch.wait();
       }
     }
 
     if (auto res = inherent_data.putData(kParachainId, parachain_inherent_data);
         res.has_error()) {
       SL_ERROR(log_, "cannot put an inherent data: {}", res.error());
-      return BabeError::CAN_NOT_PREPARE_BLOCK;
+      return BlockProductionError::CAN_NOT_PREPARE_BLOCK;
     }
 
     auto proposal_start = std::chrono::steady_clock::now();
-    // calculate babe_pre_digest
-    auto babe_pre_digest_res =
-        babePreDigest(ctx, slot_type, output, authority_index);
-    if (not babe_pre_digest_res) {
-      SL_ERROR(log_, "cannot propose a block: {}", babe_pre_digest_res.error());
-      return BabeError::CAN_NOT_PREPARE_BLOCK;
+
+    auto pre_digest_res = makePreDigest();
+    if (pre_digest_res.has_error()) {
+      SL_ERROR(log_, "cannot propose a block: {}", pre_digest_res.error());
```

Comment:
More precise would be: `cannot propose a block due to pre digest generation error: {}`